### PR TITLE
Save the CSEC data to a Special Project of the Referral

### PR DESCRIPTION
**Pull request by jimlebeau**
Save the CSEC data to a Special Project of the Referral.  CSEC data may be entered for child participants on the People and Role cards of a screening.  When the screening is promoted to a Referral the CSEC data will be saved as a 'Special Project' attached to the created Referral in CWS/CMS.

## Description

In CWS/CMS, each county will have a Special Project with the name of "S-CESC Referral".  The special project record and the processed referral is the basis of creating a Special Project Referral (special project ID and referral ID) record.  

Only one Special Project Referral record will be created per referral.

gov.ca.cwds.rest.services.cms.SpecialProjectReferralService.java class has been added to work with Special Project Referrals.

- saveSpecialProjectReferral() - to create a Special Project Referral record for the referral

- used in ParticipantService class from the processChildClient() method when the report type is CESC.

## Jira Issue link

https://osi-cwds.atlassian.net/browse/HOT-1959

## Tests
- [X] I have included unit tests 
- [ ] I have included functional tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the google code style of this project.
- [X] I have ran all tests for the project.
- [X] My code is in a stable state ready to be deployable, but not necessarily complete.
- [X] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProjectReferral.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import com.fasterxml.jackson.annotation.JsonFormat;
+import gov.ca.cwds.rest.api.Request;
+import gov.ca.cwds.rest.api.Response;
+import gov.ca.cwds.rest.api.domain.DomainChef;
+import gov.ca.cwds.rest.api.domain.DomainObject;
+import gov.ca.cwds.rest.api.domain.ReportingDomain;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * {@link DomainObject} representing an Allegation
```
Comment (igor-chornobay-engagepoint) [inline src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProjectReferral.java]: please fix representing a Special Project Referral

File: src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProject.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import gov.ca.cwds.rest.api.Request;
+import gov.ca.cwds.rest.api.Response;
+import gov.ca.cwds.rest.api.domain.DomainChef;
+import gov.ca.cwds.rest.api.domain.DomainObject;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * {@link DomainObject} representing an Allegation
```
Comment (igor-chornobay-engagepoint) [inline src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProject.java]: please fix representing a Special Project

**Comment by igor-chornobay-engagepoint** [comment]
please resolve conflicts

File: src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProjectReferral.java (hunk 1 above)
Comment (jimlebeau) [reply src/main/java/gov/ca/cwds/rest/api/domain/cms/SpecialProjectReferral.java]: got it

**Pull request merged**